test/cmocka/src/audio/drc/drc_math_test.c

```diff
@@ -0,0 +1,148 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2024 Intel Corporation.
+
+#include <math.h>
+#include <stdbool.h>
+#include <stdarg.h>
+#include <stddef.h>
+#include <setjmp.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <cmocka.h>
+
+#include <drc/drc_math.h>
+
+#define TEST_POINTS 500
+
+#define ABS_DELTA_TOLERANCE_T1 1.0e-9
+#define REL_DELTA_TOLERANCE_T1 1.4e-4
+#define ABS_DELTA_TOLERANCE_T2 7.4e-7
+#define REL_DELTA_TOLERANCE_T2 7.5e-6
+#define ABS_DELTA_TOLERANCE_T3 3.8e-4
+#define REL_DELTA_TOLERANCE_T3 1.9e-6
+
+static int32_t drc_inv_ref(int32_t x, int32_t precision_x, int32_t precision_y)
+{
+	double xf, yf, ys;
+
+	xf = (double)x / ((int64_t)1 << precision_x);
+	yf = 1 / xf;
+	ys = round(yf * ((int64_t)1 << precision_y));
+	if (ys > INT32_MAX)
+		ys = INT32_MAX;
+
+	if (ys < INT32_MIN)
+		ys = INT32_MIN;
+
+	return (int32_t)ys;
+}
+
+static void drc_math_inv_fixed_test_helper(int32_t x_start, int32_t x_end,
+					   int32_t x_step, int q_in, int q_out,
+					   double abs_delta_tolerance, double rel_delta_tolerance)
+{
+	double fx, fy1, fy2;
+	double abs_delta, rel_delta;
+	int64_t x1;
+	int32_t x, y1, y2;
+	int32_t differences = 0;
+	double abs_delta_max = 0;
+	double rel_delta_max = 0;
+
+	printf("%s: Testing q_in = %d q_out = %d in = %d:%d:%d\n", __func__,
+		q_in, q_out, x_start, x_step, x_end);
+
+	x1 = x_start;
+	while (x1 <= x_end) {
+		x = (int32_t)x1;
```

Comment:
would
```
for (x = x_start; x <= x_end; x += x_step)
```
also work? Maybe making `x` 64 bits to avoid wrapping around

Reply:
Packs three lines into one ... but as you prefer. The loop variable is 64 bits to avoid it to wrap if max int32_t value is not hit exactly. So why not have the above simple type cast code line. I like simple not very packed coding style while I see that you prefer more compact code. Since this is test code I want it to be clear, no worry about performance.

Reply:
> I like simple not very packed coding style while I see that you prefer more compact code. Since this is test code I want it to be clear, no worry about performance.

@singalsu sure, if that's your preference - fine this me. This is by no means critical. I just like to fit as much code as possible onto the screen at once, while still keeping some sane restrictions like 1 command per line and single empty lines between blocks. And without making the font too small :-) But - again - that's just my personal preference

Reply:
I was starting to change it but it got merged, next time then!